plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,464 @@
+                for site in insights_group['assuranceEntities']:
+                    if site['name'] == site_name:
+                        return site['uuid']
+
+    def get_pre_change_result(self, pcv_results, name, site_id, path,  **kwargs):
+        pcv_result = {}
+        for pcv in pcv_results:
+            if pcv.get("name") == name and pcv.get("fabricUuid") == site_id:
+                pcv_job_id = pcv.get("jobId")
+                pcv_path = '{0}/{1}'.format(path, pcv_job_id)
+                obj = self.nd.query_obj(pcv_path,  **kwargs)
+                pcv_result = obj['value']['data']
+        return pcv_result
+
+    def get_epochs(self, ig_name, site_name):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = '{0}/epochs?$size=1&$status=FINISHED&%24epochType=ONLINE%2C+OFFLINE&%24sort=-collectionTime%2C-analysisStartTime'.format(ig_base_path)
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        return obj['value']['data'][0]
+
+    def query_data(self, path):
+        obj = self.nd.query_obj(path, prefix = self.prefix)
+        return obj['value']['data']
+
+    def query_compliance_score(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/complianceScore?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        return self.query_data(path)
+
+    def query_compliance_count(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/count?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        return self.query_data(path)
+
+    def query_entry(self, path, size):
+        obj = self.nd.query_obj(path.format(size, 0), prefix=self.prefix)
+        entries = obj.get('entries')
+        if entries is None or len(entries) == 0:
+            return []
+        else:
+            pages, last_page = divmod(obj.get("totalItemsCount"), size)
+            if pages == 0 or (pages==1 and last_page==0) :
+                return entries
+            if last_page > 0 :
+                pages += 1
+            for page in range(1, pages):
+                obj = self.nd.query_obj(path.format(size, page), prefix = self.prefix)
+                entries += obj.get('entries')
+            return entries
+
+    def query_anomalies(self, ig_name, site_name, epoch_delta_job_id, epoch_choice):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        size = 100
+        path = epoch_delta_ig_path + "/individualTable?%24size={0}&%24page={1}"
+        if epoch_choice:
+            path = "{0}&epochStatus={1}".format(path, epoch_choice)
+        entries = self.query_entry(path, size)
+        return entries
+
+    def format_event_severity(self, events_severity):
```

Comment:
self not used, should this be static method or function outside ndi class?

Reply:
will improve that later when refactoring codebase